innopoints/views/application.py

```diff
@@ -128,11 +134,18 @@ def edit_application(project_id, activity_id, application_id):
     if current_user not in project.moderators and not current_user.is_admin:
         abort(401)
 
-    if project.lifetime_stage != LifetimeStage.ongoing:
-        abort(400, {'message': 'The application status may only be changed for ongoing projects.'})
+    if activity.internal:
+        if project.lifetime_stage != LifetimeStage.finalizing:
+            abort(400, {'message': 'Internal activity applications may only be edited '
+                                   'in the finalizing stage.'})
+    else:
+        if project.lifetime_stage != LifetimeStage.ongoing:
+            abort(400, {'message': 'The application may only be edited for ongoing projects.'})
 
     old_status = application.status
     if 'status' in request.json:
+        if activity.internal:
```

Comment:
Maybe not give an error if the status is not actually modified?
```suggestion
        if activity.internal and activity.status != request.json['status']:
```
It can be simpler in the frontend to just send the entire modified object instead of just the diff, and the backend picks the modified values